ajo/in-app-messages/README.md

```diff
@@ -0,0 +1,77 @@
+# In-App Messages
+Web In-App Messaging is a feature that empowers marketers with the ability to deliver personalized messages to web users. It allows them to display personalized messages on websites using modal-overlay type messages. With this new feature, marketers can effectively engage with web customers, driving user engagement, retention, and conversions.
+
+## Overview
+
+This sample is using [Adobe Experience Platform Web SDK](https://experienceleague.adobe.com/docs/experience-platform/edge/home.html) to display in-app messages.
+
+This is how an in-app message looks like:
+<img src="../../.assets/in-app-message1.png" width="800"/>
+
+## Getting Started
+
+1. Author in-app message content in Adobe Journey Optimizer.
+2. Configure the sample to use your Adobe Journey Optimizer orgid,  datastreamId and surfaceUrl. Sample configuration is in the .env file.
+3. Run the sample.
+
+## Running the sample
+Prerequisite: [install node and npm](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm).
+
+To run this sample:
+1. [Setup local SSL certificates for https](https://github.com/adobe/alloy-samples/blob/main/LocalSSLCertificateSetup.md)
+2. Clone the repository to your local machine.
+3. Open a terminal and change directory to this sample's folder.
+4. Run `npm install`
+5. Run `npm start`
+6. Open a web browser to https://localhost
+
+## How it works
+
+There are two ways to display in-app message content from Adobe Journey Optimizer.
+
+[Alloy](https://experienceleague.adobe.com/docs/experience-platform/edge/home.html) is included on the page.
+1. On page load the sendEvent command is used to fetch in-app message content.
+   example :
+
+`alloy("sendEvent", {
+  renderDecisions: true,
+  personalization {
+   decisionContext: {
+      "user": "alloy-sample"
+   }
+  }
+});`
+
+
+2. Manually trigger the rulesets evaluation using `evaluateRulesets` command.
+   example : 
+
+ `alloy("evaluateRulesets", {
+  renderDecisions: true,
+   personalization {
+     decisionContext: { 
+     "user": "button1"
```

Comment:
please use decisionContext that would make sense to the learner.  a key of `user` set to `button1` is confusing.

May I suggest changing the text within the button to something like "Buy Now!".  Then the decision context might be a key of `userAction` and value `buy_now`.  You don't have to use this example, but it is much easier to reason about to a user trying to make sense of how the decisionContext ought to be used.

If you do "Buy Now" for this button, i recommend showing the IAM presenting a coupon code.  Maybe the page load IAM could be to invite the user to subscribe to a newsletter or podcast.

I remember you had some very pretty IAMs early on where you provided custom HTML content.  They demo'd very nicely.  What do you think about making one of these have custom HTML? That way you could showcase a basic IAM and a richer customized IAM.

Also, I suggest making the buttons do something that correspond with the action.  If the button is "sign up", use a url to navigate to /sign-up.html.  You'll need to create that page, but it illustrates much better a real world use case.  Every customer will want to know how to use an IAM to get a conversion.  So, illustrating that something happens when the user clicks the call to action button is important.

Reply:
Added custom HTML content on page load.